pytools/graph.py

```diff
@@ -80,15 +80,15 @@
 
 # {{{ reverse_graph
 
-def reverse_graph(graph: GraphT) -> Dict[NodeT, FrozenSet[NodeT]]:
+def reverse_graph(graph: GraphT) -> GraphT:
     """
     Reverses a graph *graph*.
 
     :arg graph: An instance of :class:`GraphT`.
 
     :returns: A :class:`dict` representing *graph* with edges reversed.
     """
-    result: Dict[NodeT, Set[NodeT]] = {}
+    result: Dict[Type[GraphT], Set[Type[GraphT]]] = {}
```

Comment:
Without this change, my complains:

```
pytools/graph.py:91: error: Type variable "pytools.graph.NodeT" is unbound  [valid-type]
pytools/graph.py:91: note: (Hint: Use "Generic[NodeT]" or "Protocol[NodeT]" base class to bind "NodeT" inside a class)
pytools/graph.py:91: note: (Hint: Use "NodeT" in function signature to bind "NodeT" inside a function)
```

Reply:
That looks very weird.. `Type[GraphT]` should just be `dict` (i.e. the class).

I would say it's not worth complicating these if mypy isn't helpful..

Reply:
With 530a7dc, this isn't necessary anymore.